uplink/auth.py

```diff
@@ -43,9 +50,63 @@ def __call__(self, request_builder):
         request_builder.info["headers"]["Proxy-Authorization"] = self._auth_str
 
 
-class BearerToken(object):
+class ApiTokenParam(object):
+    """
+    Authorizes requests using a token or key in a query parameter.
+    Users should subclass this class to define which parameter is the token parameter.
+    """
+    param = None
+
     def __init__(self, token):
-        self._auth_str = "Bearer %s" % token
+        self._param_value = token
 
     def __call__(self, request_builder):
-        request_builder.info["headers"]["Authorization"] = self._auth_str
+        if not self.param:
+            raise NotImplementedError
+        request_builder.info["params"][self.param] = self._param_value
+
+
+class ApiTokenHeader(object):
+    """
+    Authorizes requests using a token or key in a header.
+    Users should subclass this class to define which header is the token header.
+    The subclass may also, optionally, define a token prefix (such as in BearerToken)
+    """
+    header = None
+    prefix = None
+
+    def __init__(self, token):
+        if self.prefix:
+            self._header_value = "%s %s" % (self.prefix, token)
+        else:
+            self._header_value = token
+
+    def __call__(self, request_builder):
+        if not self.header:
+            raise NotImplementedError
+        request_builder.info["headers"][self.header] = self._header_value
+
+
+class BearerToken(ApiTokenHeader):
+    header = "Authorization"
+    prefix = "Bearer"
```

Comment:
Typically, we should keep this sort of data private: I would recommend prefixing these attributes with an underscore (`_`), to align better with the libraries [subscribed style guide](http://google.github.io/styleguide/pyguide.html#3162-naming-convention)

Reply:
Sure. Works for me.